server/api/index.js

```diff
@@ -264,4 +264,58 @@ router.get("/users", (req, res) => {
   });
 });
 
+router.get("/user-roles", async (req, res) => {
+  const data = await db.getUsersByRole();
+  const newData = data.reduce((result, current) => {
+    const userId = current.user_id;
+    if (result[userId]) {
+      result[userId]["roles"].push(current.role);
+    } else {
+      result[userId] = {
+        name: current.name,
+        user_id: current.user_id,
+        email: current.email
+      };
+      result[userId]["roles"] = [current.role];
+    }
+
+    return result;
+  }, {});
+
+  res.send(Object.values(newData));
+});
+
+router.get("/user-roles/:id", async (req, res) => {
+  const userId = req.params.id;
+  const data = await db.getUserRoles(userId);
+  res.send(data);
+});
+
+// Add user roles to the user_roles table
+router.post("/user-roles", async (req, res) => {
+  const body = req.body;
+  try {
+    await db.addRoleToUser(body.user_id, body.role_id);
```

Comment:
You can for example first use your db.clearRolesByUser here and then call db.addRoleToUser.

Reply:
When I add this here `await db.clearRolesByUser(body.user_id);` it doesn't work

Reply:
Do you know why?

Reply:
It show the succesful message but doesn't add anything actually

Reply:
It's because in the frontend, your addRoleToUser is in a loop. So if you put your db.clearRolesByUser in that API, it will delete all roles every time. Instead, have an API which is to addRolesToUser (mention the "s" to roles). So you have only one API call, you pass the user_id and the list of role_id. Then in that case you can have a db.clearRolesByUser before and db.addRoleToUser after (and that db.addRoleToUser can be in a loop here). Does it make sense? Otherwise I can show you some code.

Reply:
Code would help me better understand it